src/EmitTyp.re

```diff
@@ -105,7 +105,14 @@ let rec renderTyp = (~language, ~indent=None, ~inFunType, typ) =>
       ++ (typ |> renderTyp(~language, ~indent, ~inFunType))
       ++ ")"
     }
-
+  | Tuple(innerTypes) =>
+    "["
+    ++ (
+      innerTypes
+      |> List.map(typ => typ |> renderTyp(~language, ~indent, ~inFunType))
```

Comment:
Should tuples be emitted like immutable arrays?

Reply:
It’s a good question. They’re immutable on the Reason side so I’d say yes.
Nit: typ => typ |> can be omitted: curry.

Also, TS has to is peculiar about naming types. Worth adding the examples. It might dislike the “[...]” for non-basic types.
Immutable arrays would get you out of that notation anyway.